### DISCUSSION TITLE
Support manifest generated by Vite

### DISCUSSION THREAD

**Discussion by gweesin** [verdaccio/verdaccio#4620] [Ideas]
**Is your feature request related to a problem?**

 I'm aiming to develop a verdaccio theme called 'verdaccio-theme-xxx' using Vite as the build tool. However, the current version of verdaccio doesn't support Vite's manifest handling. Perhaps we should consider adding compatibility for it.

**Describe the solution you'd like**

Verdaccio can handle Vite's manifest properly.

**Describe alternatives you've considered**

Alternatively, we could draft documentation outlining the supported manifest structures.

**Additional context**

Vite manifest structure:

```json
{
  "src/assets/react.svg": {
    "file": "assets/react-CHdo91hT.svg",
    "src": "src/assets/react.svg"
  },
  "src/main.tsx": {
    "file": "assets/index-BZjdfp4j.js",
    "name": "index",
    "src": "src/main.tsx",
    "isEntry": true,
    "css": [
      "assets/index-DiwrgTda.css"
    ],
    "assets": [
      "assets/react-CHdo91hT.svg"
    ]
  }
}
```

Webpack manifest structure:

```json
{
  "main.js": "-/static/main.e6bb24fc469d2304119c.js",
  "runtime.js": "-/static/runtime.e6bb24fc469d2304119c.js",
  "Version.js": "-/static/Version.e6bb24fc469d2304119c.js",
  "Home.js": "-/static/Home.e6bb24fc469d2304119c.js",
  "vendors.js": "-/static/vendors.e6bb24fc469d2304119c.js",
  "favicon.ico": "-/static/favicon.ico",
  "index.html": "-/static/index.html"
}
```

**Contribute to Verdaccio**

 - [x] I'm willing to research this issue 🥇

**Comment by juanpicado** [comment]
If you work on it +1, feel free to ask questions.

**Comment by gweesin** [comment]
I'm unsure why static assets need to be prefixed with `-/static`. Could you please clarify the rationale behind this?

In my local commits, I've implemented support for this feature; however, I noticed that Vite's manifest fields don't prepend the `publicPath` like Webpack does. Should this functionality be deemed essential, it might be more appropriate to await Vite's official support.

**Reply by juanpicado** [reply]
@gweesin I didn't notice your reply, just found this https://github.com/vitejs/vite/issues/16599 

>  Could you please clarify the rationale behind this?

I'll clarify that later on.

**Reply by gweesin** [reply]
The problem is that the Vite manifest is not compatible with Webpack. We hope that Vite can support a feature to expose the base configuration path in the manifest.json file so that Verdaccio can read and handle it later.
